src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+for the search result page is dynamic, and varies greatly depending on logic
+that needs to run on Search's web servers. There's currently no way for the
+service worker to replicate this logic and return cached HTML immediately—the
+best it could do is to pass along navigation requests to the backend web
+servers, which necessitates a network request. Without a service worker, this
+network request would happen immediately upon user navigation, but when a
+service worker is registered, it will (by default) always need to be started up
+and given a chance to execute its [`fetch event handlers`](https://developers.google.com/web/fundamentals/primers/service-workers/#cache_and_return_requests),
+even when there's no chance that those `fetch` handlers will do
+anything other than go to the network. The amount of time that it takes to start
+up and run the service worker code is pure overhead added on top of every
+navigation, and puts the service worker implementation at too much of a latency
+disadvantage to justify any other benefits.
+
+![An illustration of the SW startup blocking the navigation request.](no-preload.png)
+
+Additionally, the team found that, based on measuring service worker boot times
+on real-world devices, there was a wide distribution of startup times, with some
+low-end mobile devices taking almost as much time to start up the service worker
+as it might take to make the network request for the results page's HTML.
+
+**Solution**
+
+The single, most crucial feature that allowed the Google Search team to move
+ahead with their service worker launch is
+[navigation preload](https://developers.google.com/web/updates/2017/02/navigation-preload).
+Using navigation preload is a key performance win for any service worker that
+needs to use a response from the network to satisfy navigation requests. It
+provides a hint to the browser to start making the navigation request
+immediately, at the same time as the service worker starts up. As long as the
+amount of time it takes for the service worker to start up is less than the
+amount of time it takes to get a response back from the network, it means that
+there should not be any latency overhead introduced by the service worker.
+
+![An illustration of the SW startup done in parallel with the navigation request.](with-preload.png)
+
+The Search team also needed to avoid using a service worker on low-end mobile
+devices where that service worker boot time could exceed the time it would take
+to get back a network response. Since there's no hard-and-fast rule for what
+constitutes a "low-end" device, they came up with the heuristic of
+[checking the total RAM](https://developers.google.com/web/updates/2017/12/device-memory)
+installed on the device. Anything less than 2 gigabytes of memory fell into
+their low-end device category, where based on the real-world experiments they've
```

Comment:
You might want to remove 'where based on real-world experiences phrase at the end, as that seems implied. Up to you, but I think it can be inferred, and it is nice to lean the words a bit.